src/http_utils.cpp

```diff
@@ -216,6 +229,39 @@ std::string http_utils::standardize_url(const std::string& url) {
     return result;
 }
 
+const std::string http_utils::generate_random_upload_filename(const std::string& directory) {
+    std::string filename = directory + path_separator + upload_filename_template;
+    char *template_filename = strdup(filename.c_str());
+    int fd = 0;
+
+#if defined(_WIN32)
+    if (0 != _mktemp_s(template_filename, filename.size() + 1)) {
+        free(template_filename);
+        throw std::exception();
+    }
+    if (0 != _sopen_s(&fd, template_filename, _O_CREAT | _O_EXCL | _O_NOINHERIT, _SH_DENYNO, _S_IREAD | _S_IWRITE)) {
```

Comment:
if we are just trying to read the file to verify it is created correctly, we can just use `ifstream`

Reply:
That's the point where the file is actually created in the file system.
`_mktemp_s()` just generates a file name but does not create the file in the file system.

If this call fails, the file could not be created.

Of course, there are other commands to create a file, but I went for `_sopen_s()` as with this function you can specify the flags (like _O_CREAT, ...) in detail.

Reply:
If there is something highly specific we are trying to do here with the explicit params, let's keep `_sopen_s` but explain what we are trying to achieve in a comment so that it is immediately clear.
If there isn't, in general I'd prefer using C++ constructs (ofstream/ifstream) over C ones; they tend to port more easily across platforms.